### PR TITLE
docs: Many dictionary entry fixes

### LINKED ISSUE #1024: marked.js doesn't handle <button> correctly
When making a reference to the "button" dictionary entry by putting `<button>` into the markdown source of a dictionary entry, marked.js is incorrectly generating an actual HTML button.  This has some amusing results.

screen shot 2016-03-17 at 09 17 58

### PR DESCRIPTION
**Pull request by peter-b**
I noticed that in some cases the HTML of a dictionary entry could contain HTML buttons, due to incorrect handling of `<button>` entry cross-references.  I fixed them and came across a few other issues on the way.

Closes livecode/livecode-ide#1024.

### CODE REVIEW

File: docs/dictionary/keyword/menu.lcdoc [hunk 1]
```diff
@@ -26,49 +26,62 @@ If a button's <style> is set to "menu", its <menuMode> <property> determines how
...
+* Return, Enter,
+* Home, End, Escape,
+* PgUp, PgDown,
+* Insert (not on Mac OS X)
+
 On Mac OS X, these keys will be replaced with the appropriate system standard glyph.
 
-A tag can be specified as a '|' character followed by a string of ascii characters.  If such a tag is present then that tag will take the place of the menu label as the parameter to the <menuPick message>.
-A tag will only be recognised following &lt;accelerator&gt;.  To specify a menu tag without an accelerator shortcut, an empty &lt;accelerator&gt; can be specified using '/|' followed by the tag text.
+A tag can be specified as a `|` character followed by a string of ASCII characters.  If such a tag is present then that tag will take the place of the menu label as the parameter to the <menuPick message>.
```
Comment (livecodeali) [inline docs/dictionary/keyword/menu.lcdoc]: The `<menuPick message>` here should be `<menuPick> message` (and the ref changed in the references)

**Comment by peter-b** [comment]
@livecodeali Pull request updated!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 980a425

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 980a425c82b476da7b3a413427c6bcacde25ab8a

**Comment by livecode-vulcan** [comment]
😎 test success 980a425c82b476da7b3a413427c6bcacde25ab8a
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2127)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2135)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2120)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2135)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2142)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2148)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2157)

**Pull request merged**